library/options.inc.php

```diff
@@ -3475,6 +3475,47 @@ function getLayoutProperties($formtype, &$grparr, $sel = "grp_title")
     }
 }
 
+// mdsupport - No easy way to get just display data for a layout.
+// This function attempts to use data-fld to get text data for each field
+// Unlike base function, this function is designed for a single form.
+//
+function extract_layout_data($layoutid, $result1, $result2 = '')
+{
```

Comment:
Not sure I understand where this gets called from.

Reply:
@sjpadgett , this PR provides a standardized function to scrape layout data without waiting for it to become responsive.

As an example, currently portal relies on display_layout_rows call in portal/get_profile.php to display demographics. That binds portal's responsiveness to code put out by the standard function. The table/row layout also leads to fields appearing randomly thrown out based on length of contents of label and/or data columns. 

By getting a label/data pair, the caller gets to decide how the layout data will be displayed. If you decide to stick with rows layout, you can now let your own css control the responsiveness. For us, we found masonry layout works best.

Here is an example in desktop mode:
![image](https://user-images.githubusercontent.com/399047/153916874-fd1d0871-3c27-4830-ae93-eee497018314.png)


Here is that same page with smaller size - notice the insurance data as how the existing table code will show -
![image](https://user-images.githubusercontent.com/399047/153918181-f4d62011-bd7e-4dba-9a51-894d3234abea.png)